src/test/scala/is/hail/methods/AggregatorSuite.scala

```diff
@@ -262,4 +261,55 @@ class AggregatorSuite extends SparkSuite {
       .aggregate("AGG.map(r => r.foo).sum()")._1
       == Seq((0 until 100).map(_ / 2.0).sum))
   }
+
+  @Test def testLinreg() {
+    val t = Table.parallelize(hc, IndexedSeq(
+      Row(null, 1d, 0.2),
+      Row(null, 1d, null),
+      Row(0.3, 1d, null),
+      Row(0.22848042, 1d, 0.2575928),
+      Row(0.09159706, 1d, -0.3445442),
+      Row(-0.43881935, 1d, 1.6590146),
+      Row(-0.99106171, 1d, -1.1688806),
+      Row(2.12823289, 1d, 0.5587043)),
+      TStruct("y" -> TFloat64(), "intercept" -> TFloat64(), "x" -> TFloat64()),
+      None,
+      None
+    )
+
+    val (aggResult, typ) = t.aggregate("AGG.map(__uid1__ => row.y).linreg(__uid2__ => [row.intercept, row.x], 2)")
+
+    val expectedResult = Row(
+      FastIndexedSeq(0.14069227, 0.32744807),
+      FastIndexedSeq(0.59410817, 0.61833778),
+      FastIndexedSeq(0.23681254, 0.52956181),
+      FastIndexedSeq(0.82805147, 0.63310173),
+      5L)
+
+    assert(typ.valuesSimilar(aggResult, expectedResult))
+
+    val mt = t.index("col_idx")
+      .annotate("row_idx" -> "1")
+      .toMatrixTable(Array("row_idx"), Array("col_idx"), Array(), Array("intercept", "y"), Array("row_idx"))
+
+    val a = mt.annotateRowsExpr("linreg" -> "AGG.map(__uid1__ => sa.y).linreg(__uid2__ => [sa.intercept, g.x], 2)")
+      .rowsTable().select("{linreg: row.linreg}", None, None).collect()(0)(0)
+
+    assert(typ.valuesSimilar(a, expectedResult))
+
+    val mtFiltered = mt
+      .annotateColsExpr("xDefined" -> "AGG.map(__uid__ => g.x).filter(x => isDefined(x)).count() == 1.toInt64()")
+      .filterColsExpr("isDefined(sa.y) && sa.xDefined")
```

Comment:
`linear_regression` should automatically filter on missing `y` and `x` has length 1, so you should be able to just use:
```
    val mtFiltered = mt.filterColsExpr("isDefined(x[0])")
```

Reply:
I don't see how that would work. `x` doesn't mean the same thing here. It has to be in the context of the MatrixTable.